src/Transactions/BlockItem.cs

```diff
@@ -0,0 +1,27 @@
+using Concordium.Sdk.Exceptions;
+using Concordium.Sdk.Types;
+using BlockItemCase = Concordium.Grpc.V2.BlockItem.BlockItemOneofCase;
+
+namespace Concordium.Sdk.Transactions;
+
+/// <summary>A block item.</summary>
+/// <param name="TransactionHash">The hash of the block item that identifies it to the chain.</param>
+/// <param name="BlockItemType">Either a SignedAccountTransaction, CredentialDeployment or UpdateInstruction.</param>
+public sealed record BlockItem(TransactionHash TransactionHash, BlockItemType BlockItemType)
+{
+    internal static BlockItem From(Grpc.V2.BlockItem blockItem) =>
+        new(
+            TransactionHash.From(blockItem.Hash.ToString()),
+            blockItem.BlockItemCase switch
+            {
+                BlockItemCase.AccountTransaction => SignedAccountTransaction.From(blockItem.AccountTransaction),
+                BlockItemCase.CredentialDeployment => CredentialDeployment.From(blockItem.CredentialDeployment),
+                BlockItemCase.UpdateInstruction => UpdateInstruction.From(blockItem.UpdateInstruction),
+                BlockItemCase.None => throw new NotImplementedException(),
```

Comment:
No, just let the current wild card take care of it.
So essentially just remove the line with `BlockItemCase.None`

Reply:
So far I have removed the `None` case, so it should be caught by the `MissingEnumException`. I personally think that's fine since even if it does happen, it would be a bug in the node as far as I know. I don't think we need explicit errors for that.

I am open to discussion though.

Reply:
Wait the `None` case is demanded by the formatter I'll add it back with a `MissingEnumException`...

Reply:
@rasmus-kirk You are right in the `None` case should never happen and reaching the case means something went wrong in the node. I think it would be good to introduce an exception representing this, just as @schwartz-concordium suggest. Maybe something like `UnexpectedNodeResponse`.